docs/faq/ImmutableData.md

```diff
@@ -2,50 +2,154 @@
+	- React-Redux's `connect` method generates components that [shallowly check reference changes to the root state](#how-react-redux-uses-shallow-checking) with the return values from the `mapStateToProps` function to see if the wrapped components actually need to re-render.
+Such [shallow checking requires immutability](#￼redux-shallow-checking-requires-immutability) to function correctly.
 - Immutable data management ultimately makes data handling safer.
 - Time-travel debugging requires that reducers be pure functions with no side effects, so that you can correctly jump between different states.
 
-In particular, Redux's use of shallow equality checking requires immutability if the store, and any connected components, are to be updated correctly. To see why, we need to understand the difference between shallow and deep equality checking in JavaScript.
 
+<a id="redux-shallow-checking-requires-immutability"></a>
+## Why does Redux’s use of shallow equality checking require immutability?
+Redux's use of shallow equality checking requires immutability if any connected components are to be updated correctly. To see why, we need to understand the difference between shallow and deep equality checking in JavaScript.
+
+
+<a id="shallow-and-deep-equality-checking"></a>
 ### How do shallow and deep equality checking differ?
 Shallow equality checking simply checks that two different _variables_ reference the same object; in contrast, deep equality checking must check every _value_ of two objects' properties.
 
 A shallow equality check is therefore as simple (and as fast) as `a === b`, whereas a deep equality check involves a recursive traversal through the properties of two objects, comparing the value of each property at each step. 
 
 It's for this improvement in performance that Redux uses shallow equality checking.
 
-### How does Redux uses shallow equality checking?
-Redux shallowly compares the value returned from a reducer with the `state` parameter passed into it. If both objects are the same (i.e. the `state` parameter passed in references the same object as that returned by the reducer), then Redux assumes that the state has not been changed and so will leave the state tree unchanged.
 
-If, however, the returned value differs from the `state` object passed in (i.e. the `state` parameter passed in references a different object from that returned by the reducer), then Redux assumes a change has been made, and so will apply that change to the state tree.
+<a id="how-redux-uses-shallow-checking"></a>
+### How does Redux use shallow equality checking?
+Redux uses shallow equality checking in its `combineReducers` function, in order to determine whether to return a new mutated root state object, or, if no mutations have been made, the current root state object.  
+
+*If the current root state object is returned by `combineReducers`, the store will not be updated.*
```

Comment:
I think this line is still an issue.  If no changes were detected, returning the same object either means that nothing really did change, or that mutations occurred.  Either way, the returned object should reflect the current contents of the state.

I think it's probably best to just remove this line entirely.